rdt-open-reader/src/test/java/io/ona/rdt/robolectric/activity/PatientRegisterActivityTest.java

```diff
@@ -179,12 +179,19 @@ public void testSelectDrawerItemShouldSyncDataFromServer() {
 
     @Test
     public void testLatestSyncDateShouldVerifyCurrentDate() {
+        LocalBroadcastManager.getInstance(patientRegisterActivity).sendBroadcast(new Intent(PatientRegisterActivity.ACTION_UPDATE_LATEST_SYNC_DATE));
         TextView tvLatestSyncDate = patientRegisterActivity.findViewById(R.id.tv_latest_sync_date);
         String lblSync = patientRegisterActivity.getResources().getString(R.string.lbl_latest_sync);
-        Assert.assertEquals(String.format(lblSync, new SimpleDateFormat(PatientRegisterActivity.LATEST_SYNC_DATE_FORMAT, Locale.getDefault()).format(new Date(currentTimeInMillis))), tvLatestSyncDate.getText().toString());
+        Assert.assertEquals(String.format(lblSync, new SimpleDateFormat(PatientRegisterActivity.LATEST_SYNC_DATE_FORMAT, Locale.getDefault()).format(new Date(System.currentTimeMillis()))), tvLatestSyncDate.getText().toString());
```

Comment:
This may not always work `System.currentTimeMillis()` since it refers to a different timestamp from what is used to populate the label.

Reply:
I think it could be a negligible difference maybe in mills or seconds. Although, we start the comparison from minutes.

Reply:
refer to https://github.com/onaio/rdt-standard/pull/899#discussion_r591153635

Reply:
the test will fail, right?

Reply:
yes, can I persist the timestamp and then use it in Test class.

Reply:
You can add a 5-10 second window around the recorded timestamp and assert that the current time falls within the  range (recorded timestamp +/- window).

Reply:
I have made some changes you can check is that fine